collector/ksmd_linux.go

```diff
@@ -0,0 +1,99 @@
+// Copyright 2015 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// +build !noksmd
+
+package collector
+
+import (
+	"fmt"
+	"os"
+	"path"
+
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+const (
+	ksmdSubsystem = "ksmd"
+)
+
+var (
+	ksmdFiles = []string{"merge_across_nodes", "pages_shared", "pages_sharing", "pages_to_scan",
+		"pages_unshared", "pages_volatile", "run", "sleep_millisecs"}
+)
+
+type ksmdCollector struct {
+	fullScans    prometheus.Counter
+	gaugeMetrics map[string]prometheus.Gauge
+}
+
+func init() {
+	Factories["ksmd"] = NewKsmdCollector
+}
+
+// Takes a prometheus registry and returns a new Collector exposing
+// kernel/system statistics.
+func NewKsmdCollector() (Collector, error) {
+	return &ksmdCollector{
+		fullScans: prometheus.NewCounter(
+			prometheus.CounterOpts{
+				Namespace: Namespace,
+				Subsystem: ksmdSubsystem,
+				Name:      "full_scans",
```

Comment:
@brian-brazil While this is definitely a recommended practice per Prometheus docs, I thought that node_exporter was some kind of "special" exporter and followed the rule "do not change underlying OS metric name". And a quick search through the sources confirmed this thought.

If this is not true anymore and for new collectors '_total' suffix should be used for counters regardless of the underlying metrics name then, sure, I will fix that.